aspnetcore-webapi/RateLimitingDemo/Tests/ApiIntegrationTests.cs

```diff
@@ -0,0 +1,59 @@
+﻿using AspNetCoreRateLimit;
+using Microsoft.AspNetCore.Mvc.Testing;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Newtonsoft.Json;
+using RateLimitingDemo.UsAspNetCoreRateLimitPackage.Model;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace Tests
+{
+    [TestClass]
+    public class ApiIntegrationTests
+    {
+        [TestMethod]
+        public void WhenAppIsStarted_IpRateLimitOptionsAreLoadedSuccessfully()
+        {
+            using var factory = new WebApplicationFactory<Program>();
+            var options = factory.Services.GetRequiredService<IOptions<IpRateLimitOptions>>();
+            Assert.AreEqual(1, options.Value.GeneralRules.Count);
+            var generalRule = options.Value.GeneralRules[0];
+            Assert.AreEqual("GET:/products", generalRule.Endpoint);
+            Assert.AreEqual("5s", generalRule.Period);
+            Assert.AreEqual(2, generalRule.Limit);
```

Comment:
Add newlines to separate Arrange/Act/Assert. Apply to all the tests.

Reply:
done